pycloudlib/lxd/instance.py

```diff
@@ -337,28 +337,29 @@ def _wait_for_cloudinit(self, *, raise_on_failure: bool):
             When `True`, if the process waiting for cloud-init exits non-zero
             then this method will raise an `OSError`.
         """
-        if self.is_vm and raise_on_failure:
+        if self.is_vm:
             retries = [30, 45, 60, 75, 90, 105]
 
             for sleep_time in retries:
                 try:
                     super()._wait_for_cloudinit(
-                        raise_on_failure=raise_on_failure)
+                        raise_on_failure=True)
                     break
                 except OSError as e:
-                    if "Failed to connect to lxd-agent" in str(e):
-                        time.sleep(sleep_time)
-                        continue
-
                     # We are getting those types of messages when booting
                     # lxd vms. However, if we wait a moment before trying
                     # again, the error fades away. That's why we are treating
                     # it here.
-                    if "cloud-init failed to start: out: ......." in str(e):
+                    lxd_errors = [
```

Comment:
I waffled between making this error prefix list more visible like a global or a class attr `retry_on_errors` ... But since it's only used here, I guess that doesn't make much sense on this pass. If we have a more general need for different clouds to handle/retry error types we can address that then.